### DISCUSSION TITLE
In what .Net version "xOrDefault" linq methods became nullable?

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/runtime#51777] [Q&A] [answered]
Right now `SingleOrDefault` is nullable: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/Single.cs#L32

But in .Net Standard 2.1 it's not. I'm using Xamarin, so cannot use anything higher than .Net Standard 2.1, but still curious, when did "?" was added there?

![image](https://user-images.githubusercontent.com/6609929/115952917-7f046480-a4f1-11eb-8ccf-9d967bf67400.png)

**Comment by GeraudFabien** [comment, marked as answer]
Well the change was done [in this commit](https://github.com/dotnet/runtime/commit/e7204f5d6fcaca5e097ec854b3be6055229fc442#diff-5461f91894ccd633b69476a09703f4c2b857ec0d624aad13caa33f95b985f839). Release in the version [16 days ago](https://github.com/dotnet/runtime/releases/tag/v6.0.0-preview.3.21201.4)

![image](https://user-images.githubusercontent.com/13610689/115954923-e28f9180-a4f3-11eb-9306-998ccaad787e.png)

![image](https://user-images.githubusercontent.com/13610689/115955049-a4df3880-a4f4-11eb-97c8-fdd98dfed5ad.png)

**Reply by gfoidl** [reply]
I'd say it came with https://github.com/dotnet/runtime/commit/122c438e0987bf7bfd2a4a02368353d3774ff963

![grafik](https://user-images.githubusercontent.com/5755834/115955108-02738500-a4f5-11eb-8232-945f540eafbb.png)

The commit you linked changed
```diff
-[return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
-public static T SingleOrDefault<T>(this System.Collections.Immutable.ImmutableArray<T> immutableArray) { throw null; }
+public static T? SingleOrDefault<T>(this System.Collections.Immutable.ImmutableArray<T> immutableArray) { throw null; }
```

**Reply by maxkoshevoi** [reply]
Oh, so it's not even present in a stable .Net version yet? Got it, thank you
